src/components/MapView/Layers/AnalysisLayer/index.tsx

```diff
@@ -62,10 +63,16 @@ function AnalysisLayer() {
     };
   }
 
-  const property =
-    analysisData instanceof ExposedPopulationResult
-      ? (analysisData.statistic as string)
-      : 'data';
+  const property = (() => {
+    switch (true) {
+      case analysisData instanceof ExposedPopulationResult:
+        return analysisData.statistic as string;
+      case analysisData instanceof PolygonAnalysisResult:
+        return 'zonal:stat:percentage';
+      default:
+        return 'data';
+    }
+  })();
```

Comment:
I noticed the following:
![image](https://user-images.githubusercontent.com/25571542/161408032-68f48f06-a558-4929-9bc7-ec69b939172c.png)

Is this true?

Reply:
hey, that's a great catch!  I just fixed it with https://github.com/WFP-VAM/prism-frontend/pull/431/commits/915d7f157b039d841c28a1846a6d4d8c3b00d7fb.  The typescript compiler and VS Code didn't flag that for me.  Are you using another IDE or have a plugin/extension installed?  Thanks!

Reply:
I'm surprised this wasn't caught by TS; isn't there anywhere you try assign `analysisData` to a `PolygonAnalysisResult`? TS should've caught it then. Maybe this is highlighting a deeper issue, since in TS's eyes analysisData could never be a `PolygonAnalysisResult` :eyes:

I'm using WebStorm, which has a bunch of random extra checks which is quite nice in this situation!

Reply:
oh cool.  I should check out WebStorm sometime.

I think I didn't catch it earlier because I reused / updated the functions used by BaselineLayerResult rather than creating functions specific for `analysisResult`, so we never call a function where `analysisResult` needs to be a `PolygonAnalysisResult`.

Lastly, I thought it was interesting that types aren't tracked through the Redux Store.  You can pass something in as PolygonAnalysisResult, but pull it out as BaselineLayerResult.  It's not really a bug, but I'll try to keep that in mind in the future :-)

I'll resolve this now as the bug has been fixed.  Feel free to open a new one if you find any more issues.